crypto/merklearray/proof.go

```diff
@@ -96,3 +98,33 @@ func (p *SingleLeafProof) GetConcatenatedProof() []byte {
 	}
 	return proofconcat
 }
+
+// ProofDataToSingleLeafProof receives serialized proof data and uses it to construct a proof object.
+func ProofDataToSingleLeafProof(hashTypeData string, treeDepth uint64, proofBytes []byte) (SingleLeafProof, error) {
```

Comment:
should we add a unit-test that creates a singleLeafProof -> calls GetConcatenatedProof -> calls ProofDataToSingleLeafProof